desktop-widgets/diveplanner.cpp

```diff
@@ -232,10 +251,31 @@ void DivePlannerWidget::heightChanged(const int height)
 	plannerModel->setSurfacePressure(pressure);
 }
 
-void DivePlannerWidget::salinityChanged(const double salinity)
+void DivePlannerWidget::waterTypeUpdateTexts()
+{
+	double density;
+	/* Do not set text in last/custom element */
+	for (int i = 0; i < ui.waterType->count()-1; i++) {
+		if (ui.waterType->itemData(i) != QVariant::Invalid) {
+			QString densityText = ui.waterType->itemText(i).split("(")[0].trimmed();
+			density = ui.waterType->itemData(i).toInt() / 10000.0;
+			densityText.append(QString(" (%1kg/l)").arg(QString::number(density, 'f', 2)));
```

Comment:
"ℓ" also here.

Reply:
Thank you for your feedback!
> Please better no popup window!
I like the layout which looks like this one:

The version with the extra spinbox does not use pop up windows. The second gui is also my favourite.  So I think we go for that.

> When selecting one of the presets "1.00", "1.03" I would "push" the selected value also into the grayed out text box and then bring back the old custom value when "custom" is selected.

Yes, I already thought about it. I will implement it that way.

> Add 4th option which is the EN13319 value of 1,02kg/l. Some (even many?) dive computers use this value as a default.

I currently get the values from units.h. I would rather like to change the default in units.h globally for the whole program, if 1.02kg/l is more sophisticated. Otherwise we would end up having two different types of sea water in the combo box. How should we label that? "Sea water according to EN 13319 (1.02kg/ℓ)" is too long and "SW EN13319 (1.02kg/ℓ)" is too cryptic. 
Is there any good reason *not* to change the SEAWATER_SALINITY definition in units.h?

PS: I will change "kg/l" to "kg/ℓ".

Reply:
From my understanding EN13319 is not the same as sea water and for sure also not the same as fresh water. It is a kind of "compromise" used for devices where configuring a value is not possible or not requested.
This is why I would keep the SEAWATER_SALINITY with a value of 1.03 and add EN13319 simply as "EN13319 (1.02kg/ℓ)" also with a new define "EN13319_SALINITY".